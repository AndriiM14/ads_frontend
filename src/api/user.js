import axios from 'axios';
import { DEFAULT_HEADERS, formAuthHeaders } from './config';

export const createUser = (data) => axios.post('/registration', data, { headers: DEFAULT_HEADERS });

export const getBootstrap = () => axios.get('/user', { headers: formAuthHeaders() });

export const getUsers = () => axios.get('/users', { headers: formAuthHeaders() });

export const getUser = (id) => axios.get(`/users/${id}`, { headers: formAuthHeaders() });

export const deleteUser = (id) => axios.delete(`/users/${id}`, { headers: formAuthHeaders() });

export const changeUser = (id, data) => axios.put(`/users/${id}`, data, { headers: formAuthHeaders() });
