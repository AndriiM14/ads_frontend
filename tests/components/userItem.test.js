import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { mockUser } from "../mock/user";
import UserItem from "../../src/components/userItem";

describe('AdItem component tests', () => {
    const data = mockUser;

    test("Proper data render", () => {
        render(<UserItem data={data} onDelete={jest.fn()} onSelect={jest.fn()} />);

        const titleComponent = screen.getByTestId("title");
        const contentComponent = screen.getByTestId("content");
        const selectComponent = screen.getByTestId("select");
        const removeComponent = screen.getByTestId("remove");

        fireEvent(selectComponent, new MouseEvent('click'));
        fireEvent(removeComponent, new MouseEvent('click'));

        expect(titleComponent.textContent).toBe(`${data.lastname} ${data.firstname}`);
        expect(contentComponent.textContent).toBe(`${data.country} ${data.city} ${data.address}`);
    });

    test("Snapshot", async () => {
        const { asFragment } = render(<UserItem data={data} onDelete={jest.fn()} onSelect={jest.fn()} />);
        expect(asFragment).toMatchSnapshot();
    });
});