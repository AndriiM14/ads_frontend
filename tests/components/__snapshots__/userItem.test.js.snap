// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdItem component tests Snapshot 1`] = `[Function]`;
